### PR TITLE
Fix connection issues with local master if only connected usb tethering used.

### PR DESCRIPTION
**Pull request by malteim**
Fixes this this issues:
- App was crashing if local master was enabled and the phone was offline.
- App was crashing because of connection issues with parameter node (`java.lang.RuntimeException: java.net.ConnectException: failed to connect to /192.168.42.129 (port 11311): connect failed: ENETUNREACH (Network is unreachable)`).

### CODE REVIEW

File: TangoRosStreamer/app/src/main/java/eu/intermodalics/tango_ros_streamer/activities/RunningActivity.java [hunk 1]
```diff
@@ -385,9 +385,13 @@ public void run() {
                     mSnackbarLoadNewMap.setAction(getString(R.string.snackbar_action_text_load_new_map), new View.OnClickListener() {
                         @Override
                         public void onClick(View view) {
-                            mParameterNode.changeSettingsToLocalizeInMap(mapUuid, getString(R.string.pref_create_new_map_key),
-                                    getString(R.string.pref_localization_mode_key), getString(R.string.pref_localization_map_uuid_key));
-                            restartTango();
+                            try {
```
Comment (PerrineAguiar) [inline TangoRosStreamer/app/src/main/java/eu/intermodalics/tango_ros_streamer/activities/RunningActivity.java]: I think you also need to add a try/catch block here: https://github.com/Intermodalics/tango_ros/pull/359/files#diff-a79d89dbb3540e8c2b65c1f9155800a7R344

File: TangoRosStreamer/app/src/main/java/eu/intermodalics/tango_ros_streamer/activities/RunningActivity.java [hunk 2]
```diff
@@ -550,7 +564,16 @@ public boolean onCreateOptionsMenu(Menu menu) {
     public boolean onOptionsItemSelected(MenuItem item) {
         switch (item.getItemId()) {
             case R.id.settings:
-                if(mParameterNode != null) mParameterNode.setPreferencesFromParameterServer();
+                if(mParameterNode != null) {
+                    try {
+                        mParameterNode.setPreferencesFromParameterServer();
+                    } catch (Exception e) {
+                        // java.lang.RuntimeException: java.net.ConnectException:
```
Comment (PerrineAguiar) [inline TangoRosStreamer/app/src/main/java/eu/intermodalics/tango_ros_streamer/activities/RunningActivity.java]: Cleanup commented code.

File: TangoRosStreamer/app/src/main/java/eu/intermodalics/tango_ros_streamer/activities/RunningActivity.java [hunk 3]
```diff
@@ -589,7 +618,14 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
             if ((requestCode == StartSettingsActivityRequest.STANDARD_RUN ||
                     requestCode == StartSettingsActivityRequest.FIRST_RUN) &&
                     mParameterNode != null) {
-                mParameterNode.uploadPreferencesToParameterServer();
+                try {
+                    mParameterNode.uploadPreferencesToParameterServer();
+                } catch (Exception e) {
+                    // java.lang.RuntimeException: java.net.ConnectException:
```
Comment (PerrineAguiar) [inline TangoRosStreamer/app/src/main/java/eu/intermodalics/tango_ros_streamer/activities/RunningActivity.java]: Same here.

**Comment by PerrineAguiar** [comment]
test this please

**Comment by PerrineAguiar** [comment]
@malteim Thanks again, I fixed the remaining issues.

**Comment by meyerj** [comment]
@malteim Small remark: If I enable local master and there are no active network connections (no Wifi or USB tethering enabled) the app shows the last configured external `ROS_MASTER_URI`. Not really a problem as there is no way to communicate from the outside. But either the app should show `http://localhost:11311` as the master URI or simply exit with an error message.

**Comment by PerrineAguiar** [comment]
@meyerj The app does show an error message. But it is true that the user can miss it since it is displayed as a toast message.

**Pull request merged**